boxoffice/models/discount_policy.py

```diff
@@ -78,6 +79,15 @@ def __init__(self, *args, **kwargs):
         self.secret = kwargs.get('secret') if kwargs.get('secret') else buid()
         super(DiscountPolicy, self).__init__(*args, **kwargs)
 
+    @validates('discount_code_base')
+    def validate_discount_code_base(self, key, value):
+        assert not DiscountPolicy.query.filter(
+            DiscountPolicy.id != self.id,
+            DiscountPolicy.organization == self.parent,
+            DiscountPolicy.discount_code_base == value
+        ).notempty()
+        return value
```

Comment:
This is not required. The form validator is sufficient. This validator will trigger a SQL flush in the middle of a transaction, which you don't want.